src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,52 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.producer.BulldozerProducer;
+import core.mate.academy.model.producer.ExcavatorProducer;
+import core.mate.academy.model.producer.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    public static final String BULLDOZER = "core.mate.academy.model.Bulldozer";
+    public static final String TRUCK = "core.mate.academy.model.Truck";
+    public static final String EXCAVATOR = "core.mate.academy.model.Excavator";
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        MachineProducer<? extends Machine> producer;
+
+        switch (type.getName()) {
+            case BULLDOZER:
+                producer = new BulldozerProducer();
+                break;
+            case TRUCK:
+                producer = new TruckProducer();
+                break;
+            case EXCAVATOR:
+                producer = new ExcavatorProducer();
+                break;
+            default:
+                return new ArrayList<>();
+        }
+
+        return (List<Machine>) producer.get();
+    }
+
+    @Override
+    public void fill(List<? super Machine> machines, Machine value) {
+        for (int i = 0; i < machines.size(); i++) {
+            machines.set(i, value);
+        }
```

Comment:
Just found out that the method Collections.fill(machines, value); will do the same.

Reply:
`machines.replaceAll(ignored -> value);`
Do the same too, but i decided to make it by myself 😊